packages/frontend-api/src/Resources/config/graphql-types/Customer/DeliveryAddress/DeliveryAddressInputObjectDecorator.types.yaml

```diff
@@ -0,0 +1,82 @@
+                description: "Delivery address first name"
+                validation:
+                    -   NotBlank:
+                            message: "Please enter first name"
+                    -   Length:
+                            max: 100
+                            maxMessage: "First name cannot be longer than {{ limit }} characters"
+            lastName:
+                type: "String!"
+                description: "Delivery address last name"
+                validation:
+                    -   NotBlank:
+                            message: "Please enter last name"
+                    -   Length:
+                            max: 100
+                            maxMessage: "Last name cannot be longer than {{ limit }} characters"
+            companyName:
+                type: "String"
+                description: "Delivery address company name"
+                defaultValue: null
+                validation:
+                    -   Length:
+                            max: 100
+                            maxMessage: "Company name cannot be longer than {{ limit }} characters"
+            street:
+                type: "String!"
+                description: "Delivery address street name"
+                validation:
+                    -   NotBlank:
+                            message: "Please enter street"
+                    -   Length:
+                            max: 100
+                            maxMessage: "Street name cannot be longer than {{ limit }} characters"
+            city:
+                type: "String!"
+                description: "Delivery address city name"
+                validation:
+                    -   NotBlank:
+                            message: "Please enter city"
+                    -   Length:
+                            max: 100
+                            maxMessage: "City name cannot be longer than {{ limit }} characters"
+            postcode:
+                type: "String!"
+                description: "Delivery address zip code"
+                validation:
+                    -   NotBlank:
+                            message: "Please enter zip code"
+                    -   Length:
+                            max: 30
+                            maxMessage: "Zip code cannot be longer than {{ limit }} characters"
+            telephone:
+                type: "String"
+                description: "Delivery address telephone"
+                defaultValue: null
+                validation:
+                    -   Length:
+                            min: 9
+                            minMessage: "Telephone number cannot be shorter than {{ limit }} characters"
+                            max: 30
+                            maxMessage: "Telephone number cannot be longer than {{ limit }} characters"
+                    -   Regex:
+                            pattern: '/^[0-9\+]+$/' # see \App\Component\Validator\RegexValidationRule::TELEPHONE_REGEX
```

Comment:
Fixed. This comment was used also in TelephoneInputObjectDecorator (https://github.com/shopsys/shopsys/pull/3265/commits/5dad25e64b70eaddf49921f8e1af946c411da7dd), I removed it from there also.
And that raised a question, shouldn't we use `TelephoneInputObjectDecorator` ? in `DeliveryAddressInputObjectDecorator`. There is a little problem, that phone is not mandatory for delivery address.

Reply:
that would be awesome, however, I do not know whether we are able to somehow alter the validations :thinking: We can try it together, let me know :wink:

Reply:
One thing comes to my mind, using groups in input objects. In specific mutation you should define correct validation group(s). 
It is not an easy task to turn off validation from package `fronted-api` on project. Thanks to that, it would be easier to mange validations on projects. Developers simply override mutation and don't need to change definition of GQL input-object

e.g.
```
TelephoneInputObjectDecorator:
    type: input-object
    decorator: true
    config:
        fields:
            telephone:
                type: "String!"
                description: "The customer's telephone number"
                validation:
                    -   NotBlank:
                            message: "Please enter telephone number"
                            validationGroup: ["RegistrationMutation", "OrderMutation"]
                    -   Length:
                            min: 9
                            minMessage: "Telephone number cannot be shorter than {{ limit }} characters"
                            max: 30
                            maxMessage: "Telephone number cannot be longer than {{ limit }} characters"
                            validationGroup: ["RegistrationMutation"]
                    -   Regex:
                            pattern: '/^[0-9\+]+$/'
                            message: "Please enter only numbers and the + sign"
```
then
in `SSFW\CustomerUserMutation::registerMutation`
```
    public function registerMutation(Argument $argument, InputValidator $validator): array
    {
        $validator->validate(['Default', 'RegistrationMutation']);
    }
```
on project, I don't want a telephone to be mandatory, so I override `App\CustomerUserMutation::registerMutation`
```
    public function registerMutation(Argument $argument, InputValidator $validator): array
    {
        $validator->validate();
    }
```

There is also a "BUT" in the solution. But what should I do, when I don't want telephone to be mandatory, but Lenght validation should be preserved. :upside_down_face: :gun: